lib/ddtrace/writer.rb

```diff
@@ -29,53 +31,63 @@ def initialize(options = {})
 
       @traces_flushed = 0
 
-      # spawns two different workers for spans and services;
-      # they share the same transport which is thread-safe
-      @trace_worker = nil
-      @service_worker = nil
+      # one worker for both services and traces, each have their own queues
+      @worker = nil
     end
 
     # spawns two different workers for spans and services;
     # they share the same transport which is thread-safe
     def start
-      @trace_worker = Datadog::Workers::AsyncTransport.new(@span_interval, @transport, @buff_size) do |items, transport|
-        send_spans(items, transport)
-      end
-      @service_worker = Datadog::Workers::AsyncTransport.new(@service_interval, @transport, 1) do |items, transport|
-        send_services(items, transport)
+      @next_send_services = Time.now
+      @worker = Datadog::Workers::AsyncTransport.new(@span_interval,
+                                                     @transport,
+                                                     [:traces, :services],
+                                                     @buff_size) do |items, transport|
+        send(items, transport)
       end
 
-      @trace_worker.start()
-      @service_worker.start()
+      @worker.start()
     end
 
     # stops both workers for spans and services.
     def stop
-      @trace_worker.stop()
-      @trace_worker = nil
-      @service_worker.stop()
-      @service_worker = nil
+      @worker.stop()
+      @worker = nil
+    end
+
+    # flush spans to the trace-agent, handles both spans & services
+    def send(items, transport)
```

Comment:
After other changes, we may think if this par should stay here, in the ``AsyncTransport`` or if we can merge more stuff in one single ``write`` method.